dts/bindings/charger/nxp,pf1550-charger.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright (c), 2024 Arduino SA
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP PF1550 battery charger
+
+include: battery.yaml
+
+compatible: "nxp,pf1550-charger"
+
+properties:
+  constant-charge-voltage-max-microvolt:
+    required: true
+
+  constant-charge-current-max-microamp:
+    required: true
+
+  pf1550,vbus-current-limit-microamp:
+    type: int
+    required: true
+    description: |
+      VBUS current limit in microamperes.
+
+  pf1550,system-voltage-min-threshold-microvolt:
+    type: int
+    required: true
+    enum:
+      - 3500000
+      - 3700000
+      - 4300000
+    description: |
+      System voltage minimum threshold.
+
+  pf1550,thermistor-monitoring-mode:
+    type: string
+    required: true
+    enum:
+      - "disabled"
+      - "thermistor"
+      - "JEITA-1"
+      - "JEITA-2"
+    description: |
+      Thermistor monitoring mode.
+      Refer to ThrmCfg register description and Table 2 for details.
+
+  pf1550,int-gpios:
+    type: phandle-array
+    required: true
+    description: Interrupt pin
+
+  #  The following macros are available:
+  #  - LED_ON_IN_CHARGING_FLASH_IN_FAULT
+  #  - LED_FLASH_IN_CHARGING_ON_IN_FAULT
+  #  - LED_MANUAL_OFF
+  pf1550,led-behaviour:
```

Comment:
should likely be a string property, see e.g. how nPM1300 LED behavior is specified. You can later use DT_ENUM_IDX helpers to get a number.